### ISSUE TITLE
CLI boolean flag regression (was: CLI brace expansion regression)

### ISSUE THREAD

**Issue by emmenko** [prettier/prettier#2683]
Hi guys,

I'm upgrading to `v1.6.0` and I've noticed that the command using glob patterns does not work anymore (it just exists with an error). Is it just me or you also encountered this problem?

* with brace expansion: `'{lib,test}/**/*.js'`
```bash
$ yarn format
yarn format v0.27.5
$ prettier --write --trailing-comma es5 --single-quote '{lib,test}/**/*.js'
Usage: prettier [opts] [filename ...]

Available options:
  --write                  Edit the file in-place. (Beware!)
  --list-different or -l   Print filenames of files that are different from Prettier formatting.
  --config                 Path to a prettier configuration file (.prettierrc, package.json, prettier.config.js).
  --no-config              Do not look for a configuration file.
  --find-config-path <path>
                           Finds and prints the path to a configuration file for a given input file.
  --ignore-path <path>     Path to a file containing patterns that describe files to ignore.
                           Defaults to ./.prettierignore.
  --stdin                  Read input from stdin.
  --stdin-filepath         Path to the file used to read from stdin.
  --print-width <int>      Specify the length of line that the printer will wrap on. Defaults to 80.
  --tab-width <int>        Specify the number of spaces per indentation-level. Defaults to 2.
  --use-tabs               Indent lines with tabs instead of spaces.
  --no-semi                Do not print semicolons, except at the beginning of lines which may need them.
  --single-quote           Use single quotes instead of double quotes.
  --no-bracket-spacing     Do not print spaces between brackets.
  --jsx-bracket-same-line  Put > on the last line instead of at a new line.
  --trailing-comma <none|es5|all>
                           Print trailing commas wherever possible when multi-line. Defaults to none.
  --parser <flow|babylon|typescript|postcss|json|graphql>
                           Specify which parse to use. Defaults to babylon.
  --cursor-offset <int>    Print (to stderr) where a cursor at the given position would move to after formatting.
                           This option cannot be used with --range-start and --range-end
  --range-start <int>      Format code starting at a given character offset.
                           The range will extend backwards to the start of the first line containing the selected statement.
                           This option cannot be used with --cursor-offset.
                           Defaults to 0.
  --range-end <int>        Format code ending at a given character offset (exclusive).
                           The range will extend forwards to the end of the selected statement.
                           This option cannot be used with --cursor-offset.
                           Defaults to Infinity.
  --no-color               Do not colorize error messages.
  --with-node-modules      Process files inside `node_modules` directory.
  --version or -v          Print Prettier version.


error Command failed with exit code 1.
```

* without brace expansion: `'lib/**/*.js' 'test/**/*.js'`

```bash
$ yarn format
yarn format v0.27.5
$ prettier --write --trailing-comma es5 --single-quote 'lib/**/*.js' 'test/**/*.js'
test/api.spec.js 112ms
...
Done in 0.90s.
```

Thanks for the help and for this awesome library! 🙌

**Comment by emmenko** [comment]
Hmm, weird enough, if I run it for `md` and `css` files it works 🤔

```bash
$ yarn run format:css
yarn run v0.27.5
$ prettier --parser postcss --write '{lib,modules,plugins}/**/*.css'
...
Done in 2.74s.

$ yarn run format:md
yarn run v0.27.5
$ pmd *.md {docs,lib,modules,plugins}/**/*.md
PrettierMarkdown: start
PrettierMarkdown: finished.

- success: README.md
...
Done in 0.63s.
```

Something wrong with the `js` files? Did I miss something?

**Comment by emmenko** [comment]
Ok, I figured it out. The glob patterns needs to be passed right after the `--write` option. Not sure why or if it's intended this way, I guess it wasn't clear enough from the documentation.

Sorry about that.

**Comment by lydell** [comment]
No, you are correct that this is a regression. The problem boils down to:

```
prettier <any-boolean-flag> some-file.js
```

The above is supposed to print some-file.js prettified to stdout, but is instead showing the help. For example:

```
prettier --single-quote some-file.js
```

`--use-tabs`, `--jsx-bracket-same-line` etc have the same problem. Interestingly, `--no-semi` and `--no-bracket-spacing` work, but not `--semi` and `--bracket-spacing`.

**Comment by emmenko** [comment]
I see, thanks for looking into it so quickly 😄

**Comment by vjeux** [comment]
I just shipped 1.6.1 that contains this fix! Thanks for reporting it!

**Comment by emmenko** [comment]
That was fast 😄 
Thanks for the quick fix! 🙌
